### PR TITLE
Position anchor link so that it doesn't cause header text to dance

### PR DESCRIPTION
**Pull request by develohpanda**
### Summary

This PR updates the CSS for showing anchor links. This PR also makes the anchor a tiny bit smaller, but we can alter the size as needed (see comments in CSS).

### Reason

Hovering on headers that would wrap onto multiple lines would cause the header to dance around as the anchor changed between hidden and visible. This PR takes inspiration from [GitHub Actions docs](https://docs.github.com/en/actions/creating-actions/metadata-syntax-for-github-actions#about-yaml-syntax-for-github-actions) to fix up the styling.

### Testing

This header shouldn't dance: https://insomnia-docs-git-funky-anchor-padding-teaminsomnia.vercel.app//insomnia/hooks-and-actions#example-plugin-to-get-request-details-in-a-modal

### CODE REVIEW

File: docs/_includes/anchor-links.html [hunk 1]
```diff
@@ -1,9 +1,9 @@
 <script>
   let anchorForId = function (id) {
     let anchor = document.createElement("a");
-    anchor.className = "header-link";
+    anchor.className = "anchor-link";
     anchor.href      = "#" + id;
-    anchor.innerHTML = "<span class=\"sr-only\"></span><img class=\"anchor-link\" src=\"/assets/images/anchor-link.svg\" alt=\"\"></img>";
+    anchor.innerHTML = "<span class=\"sr-only\"></span><img class=\"anchor-icon\" src=\"/assets/images/anchor-icon.svg\" alt=\"\"></img>";
```
Comment (develohpanda) [inline docs/_includes/anchor-links.html:6]: What does the `sr-only` class on the `span` do? I understand that it's for screen readers, but not sure of its specific purpose here. Is it supposed to hide the icon?

Comment (falondarville) [reply docs/_includes/anchor-links.html:6]:
```suggestion
    anchor.innerHTML = "<span class=\"sr-only\">Permalink</span><img class=\"anchor-icon\" src=\"/assets/images/anchor-icon.svg\" alt=\"\"></img>";
```

Good catch. I was missing screenreader text information.

**Review by falondarville — APPROVED** [review]
LGTM. Thanks for handling.

File: docs/_includes/anchor-links.html (hunk 1 above)
Comment (develohpanda) [reply docs/_includes/anchor-links.html]: Oh, that explains everything!

**Pull request merged**